src/test/java/org/openrewrite/java/migrate/guava/PreferJavaUtilOptionalTest.java

```diff
@@ -0,0 +1,309 @@
+                  }
+              }
+              """, """
+              import java.util.Optional;
+
+              class A {
+                  String foo(Optional<String> optional) {
+                      return optional.orElse(null);
+                  }
+              }
+              """));
+    }
+
+    @Test
+    void orToOrElse() {
+        //language=java
+        rewriteRun(java("""
+          import com.google.common.base.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.or("other");
+              }
+          }
+          """, """
+          import java.util.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.orElse("other");
+              }
+          }
+          """));
+    }
+
+    @Test
+    void orSupplierToOrElseGet() {
+        //language=java
+        rewriteRun(java("""
+          import com.google.common.base.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.or(() -> "other");
+              }
+          }
+          """, """
+          import java.util.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.orElseGet(() -> "other");
+              }
+          }
+          """));
+    }
+
+    @Test
+    void transformToMap() {
+        //language=java
+        rewriteRun(java("""
+          import com.google.common.base.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.transform(String::toUpperCase);
+              }
+          }
+          """, """
+          import java.util.Optional;
+
+          class A {
+              Optional<String> foo(Optional<String> optional) {
+                  return optional.map(String::toUpperCase);
+              }
+          }
+          """));
+    }
+
+    @Nested
+    class NotYetImplemented {
+        @Test
+        @ExpectedToFail("Not yet implemented")
+        void orOptionalToTernary() {
+            // Comparison to java.util.Optional: this method has no equivalent in Java 8's Optional class; write thisOptional.isPresent() ? thisOptional : secondChoice instead.
+            //language=java
+            rewriteRun(java("""
+              import com.google.common.base.Optional;
+
+              class A {
+                  Optional<String> foo(Optional<String> firstChoice, Optional<String> secondChoice) {
+                      return firstChoice.or(secondChoice);
+                  }
+              }
+              """, """
+              import java.util.Optional;
+
+              class A {
+                  Optional<String> foo(Optional<String> firstChoice, Optional<String> secondChoice) {
+                      return a.isPresent()? firstChoice : secondChoice;
```

Comment:
There's probably a typo: 
```suggestion
                      return firstChoice.isPresent()? firstChoice : secondChoice;
```
However I think this would be way more idiomatic that way (requires java >= 9):
```suggestion
                      return firstChoice.or(() -> secondChoice);
```

Reply:
Thanks for spotting the issue and the improvement suggestion; the [docs specifically mentioned the predicate replacement](https://guava.dev/releases/31.1-jre/api/docs/com/google/common/base/Optional.html#or(com.google.common.base.Optional)). Now I'm trying to think if there's any side effects that might occur with `.or(() -> ...)`, but can't yet think of any. Would you agree there?

Reply:
Using `or()` is certainly better, because the ternary operator would either require introducing a variable or evaluate the first optional expression twice, unless it is a plain variable reference. I think there is also a convoluted alternative for Java 8.

Reply:
Does remind me we should add a restriction to only run these on Java 9+...